semgrep_output_v1.atd

```diff
@@ -2167,6 +2169,37 @@ type resolution_result = [
   | ResolutionError of resolution_error list
 ]
 
+(* ----------------------------- *)
+(* SCA part 4: Symbol analysis *)
+(* ----------------------------- *)
+
+(* "Symbol analysis" is about determining the third-party functions which
+   are used by a given project.
+
+   We may be interested in this for the purposes of analyzing the vulnerability
+   of the repo based on its dependencies, and providing more customized/informed
+   guidance on how to remediate SCA vulns.
+
+   For instance, source code like
+   ```
+   from werkzeug.wrappers import Request
+   ```
+   may be described as using the "symbol" [werkzeug, wrappers, Request].
+ *)
+
+(* A symbol is a FQN. *)
+type symbol = string list
```

Comment:
I would make it a record so it can be changed later in a backward compatible way (ATD has good support
for adding fields in a record and be backward compatible; not so much for other types).
type symbol = {
  fully_qualified_name: string list;
}

Reply:
or maybe fqn: string list;
to save some space in the sent JSON.